### DISCUSSION TITLE
Commit to a private repo

### DISCUSSION THREAD

**Discussion by TasteLook** [community/community#31713] [New to GitHub]
Can a user not added as a collaborator in a private repo make commits and changes to it?

**Comment by airtower-luna** [comment]
Not directly. There's nothing that'd prevent someone with write access from pushing a commit authored by someone who doesn't have access, though.

**Reply by TasteLook** [reply]
Thank you

**Comment by PythonCoderAS** [comment]
No, only someone with collaborator access can push content there. Nothing stopping that person from accepting commits from other people though.

**Reply by TasteLook** [reply]
Thank you

**Comment by Kirtishukla2004** [comment]
Commits can be authored, committed, and pushed by different users. Non-collaborators can't push into your repository, but they can definitely provide commits for a collaborator to push.

**Reply by TasteLook** [reply]
Thank you

**Reply by Kirtishukla2004** [reply]
i will be happy if you will mark this as answerd
